python/mp/rosary.py

```diff
@@ -338,11 +339,15 @@ def add_effect(self, *args, **kwargs):
         # it's all the same to us
         kwargs['bead_set'] = bead_set
         kwargs['color'] = effect_color
+        kwargs['bead_set_sort'] = bead_set_sort
 
         # I'd rather be fancy and strip out kwargs that won't be accepted
         # than force people writing effects to take **kwargs /flex
         requested_effect = self.effect_registry.get(effect_name)
         requested_effect_args = inspect.getargspec(requested_effect).args
+        # I don't want to add this to all the effects that are already written, but this
+        # solution feels like a gross hack
+        requested_effect_args.append('bead_set_sort')
 
```

Comment:
Just like the comment says, this is my solution so I don't have to add bead_set_sort to all the effect objects.
I think the way we should handle this is with a list of "default arguments" that are always passed, that get appended to the arguments discovered by inspection.